### PR TITLE
feat: step prev event

### PR DESCRIPTION
**Pull request by csiyang**
# Description

### <samp>🤖 Generated by Copilot at 6407a28</samp>

This pull request adds support for tracking and creating step prev events, which are events that occur when a user navigates to a previous step in a journey. It modifies the GraphQL schema, the database schema, the resolver classes, and the tests to handle the new event type. It also fixes some minor issues and adds comments in the existing code.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/34951357/todos/6733056285)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

### <samp>🤖 Generated by Copilot at 6407a28</samp>

*  Add a new column `prevStepId` to the `Event` table and a new field `prevStepId` to the `Event` model to store the id of the previous step block in a journey for a step event ([link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-3414f765c008fb078ca2038a887255fce5b7b9c1549d2b945e5e60663670dddaR1-R2), [link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-881378863f13d5e9573c2c2e756ae61e8161918ebb4a384a8d5dddd7bf7385b0R103))
*  Add a new input type `StepPrevEventCreateInput`, a new type `StepPrevEvent`, and a new mutation `stepPrevEventCreate` to the GraphQL schema in `step.graphql` to allow creating and querying step prev events ([link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-29131681a98f41397b30e90886c80b6fc28d42df8c3c5b7df71856a044d9b5abL73-R125))
*  Generate new classes `StepPrevEventCreateInput` and `StepPrevEvent` in `graphql.ts` based on the GraphQL schema changes ([link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R508-R515), [link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R1030-R1038))
*  Add a new class `StepPrevEventResolver` in `step.resolver.ts` with a method `stepPrevEventCreate` that uses the event service to validate and save the input and returns the created step prev event ([link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-b62940f4fc34e8998562c7e78b1e73dde6b81d243daa3516e0ce387838766d21R117-R145))
*  Add a new describe block in `step.resolver.spec.ts` to test the `StepPrevEventResolver` class and its method `stepPrevEventCreate` using mock inputs and expected outputs ([link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-6712a7ff1aabb2ad0ad5a51f78073ecba5226d36a065c83b0541cfd445e59dd0R163-R195))
*  Fix a typo and add comments to the `blockId` and `nextStepId` fields in the `StepViewEventCreateInput` and `StepNextEventCreateInput` input types in `step.graphql` to clarify their meanings ([link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-29131681a98f41397b30e90886c80b6fc28d42df8c3c5b7df71856a044d9b5abR6-R8), [link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-29131681a98f41397b30e90886c80b6fc28d42df8c3c5b7df71856a044d9b5abL38-R46))
*  Import the `StepPrevEventCreateInput` and `StepPrevEventResolver` classes from `graphql.ts` and `step.resolver.ts` respectively to `step.resolver.spec.ts` for testing purposes ([link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-6712a7ff1aabb2ad0ad5a51f78073ecba5226d36a065c83b0541cfd445e59dd0R5), [link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-6712a7ff1aabb2ad0ad5a51f78073ecba5226d36a065c83b0541cfd445e59dd0L10-R15))
*  Import the `StepPrevEvent` and `StepPrevEventCreateInput` classes from `graphql.ts` to `step.resolver.ts` for defining the return type and the input type for the `stepPrevEventCreate` method ([link](https://github.com/JesusFilm/core/pull/2022/files?diff=unified&w=0#diff-b62940f4fc34e8998562c7e78b1e73dde6b81d243daa3516e0ce387838766d21R12-R13))

### CODE REVIEW

File: apps/api-journeys/src/app/modules/event/step/step.graphql [hunk 1]
```diff
@@ -35,9 +38,12 @@ input StepNextEventCreateInput {
   ID should be unique Event UUID (Provided for optimistic mutation result matching)
   """
   id: ID
+  """
+  Id of the StepBlock
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/event/step/step.graphql:42]:
```suggestion
  Id of the current StepBlock
```

File: apps/api-journeys/src/app/modules/event/step/step.graphql [hunk 2]
```diff
@@ -70,7 +76,51 @@ type StepNextEvent implements Event {
   value: String
 }
 
+input StepPrevEventCreateInput {
+  """
+  ID should be unique Event UUID (Provided for optimistic mutation result matching)
+  """
+  id: ID
+  """
+  Id of the StepBlock
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/event/step/step.graphql:85]:
```suggestion
  Id of the current StepBlock
```

File: apps/api-journeys/src/app/modules/event/step/step.graphql [hunk 3]
```diff
@@ -70,7 +76,51 @@ type StepNextEvent implements Event {
...
+  """
+  ID of the journey that the stepBlock belongs to
+  """
+  journeyId: ID!
+  """
+  time event was created
+  """
+  createdAt: DateTime!
+  """
+  stepName of the stepBlock
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/event/step/step.graphql:113]:
```suggestion
  stepName of the current stepBlock
```

File: apps/api-journeys/src/app/modules/event/step/step.graphql [hunk 4]
```diff
@@ -70,7 +76,51 @@ type StepNextEvent implements Event {
...
+  """
+  Id of the StepBlock
+  """
+  blockId: ID!
+  """
+  id of the previous stepBlock
+  """
+  prevStepId: ID!
+  """
+  stepName of the stepBlock
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/event/step/step.graphql:93]:
```suggestion
  stepName of the current stepBlock
```

**Review by tataihono — DISMISSED** [review]
this looks fine. Can you change Prev to Previous in all instances. Shorthand variables names increase comprehension time.

**Review by storyworks — APPROVED** [review]

**Pull request merged**